cpu/native/socket_zep/socket_zep.c

```diff
@@ -428,14 +431,17 @@ void socket_zep_setup(socket_zep_t *dev, const socket_zep_params_t *params, uint
         dev->sock_fd = res;
     }
 
-    if (_connect_remote(dev, params) == 0) {
-        /* send dummy data to connect to dispatcher */
-        _send_zep_hello(dev);
-    }
+    /* only send hello if we are connected to a remote */
+    bool send_hello = !_connect_remote(dev, params);
 
     /* setup hardware address */
     netdev_ieee802154_setup(&dev->netdev);
 
+    /* send dummy data to connect to dispatcher */
+    if (send_hello) {
+        _send_zep_hello(dev);
+    }
+
```

Comment:
Is there a reason for the re-ordering?

Reply:
`_send_zep_hello()` now includes the hw address so it must happen after `netdev_ieee802154_setup()`